src/ex4/client/src/components/ItemComp.jsx

```diff
@@ -0,0 +1,87 @@
+import "../App.css";
+import "../style/itemComp.css";
+import { useState } from "react";
+import ButtonComp from "./ButtonComp";
+import Checkbox from "monday-ui-react-core/dist/Checkbox";
+import "monday-ui-react-core/dist/main.css";
+import trashImg from "../images/trash.png";
+import plusImg from "../images/plus.png";
+import backArrow from "../images/backArrow.png";
+
+function ItemComp({ item, deleteItem, updateItem }) {
+	const [isEditElement, setIsEditElement] = useState(false);
+	const [newItem, setNewItem] = useState(item);
+
+	const editInputAndButton = () => {
+		return (
+			<div>
+				<input
+					className="clickable, task-small-input"
+					onChange={(e) => setNewItem({ ...item, itemName: e.target.value })}
+					value={newItem.itemName}
+				/>
+				<ButtonComp
+					className={"clickable pluse-icon-element"}
+					imgSrc={backArrow}
+					onClick={() => {
+						setIsEditElement(false);
```

Comment:
note that in this case the "newItem" is still saved in the state, and next time you're entering edit it will show the value edited before